kore/src/Kore/Equation/Simplification.hs

```diff
@@ -74,22 +76,24 @@ argument contain a predicate which is not 'Top', 'simplifyEquation'
 returns the original equation.
 
  -}
+
 simplifyEquation
     :: (InternalVariable variable, MonadSimplify simplifier)
     => Equation variable
     -> simplifier (MultiAnd (Equation variable))
-simplifyEquation equation@(Equation _ _ _ _ _ _ _) =
+simplifyEquation equation =
     do
-        simplifiedResults <-
-            simplifyPattern leftWithArgument
-        Monad.when
-            (any (not . isTop . predicate) simplifiedResults)
+        simplifiedCond <- lift $
+            Simplifier.simplifyCondition
+                SideCondition.top
+                (fromPredicate argument')
+        Monad.unless
+            ((isTop . predicate) simplifiedCond)
```

Comment:
The problem is this: Before, we would return early if any of the `simplifiedResults` (i.e. any branch) had a predicate which was not `\top`. Now, we only return early _on that branch_. If you swap the order of `Logic.observeAllT` and `returnOriginalIfAborted` below (changing the order of nesting `LogicT` and `ExceptT`) then all will be well: the entire computation will exit early if any branch exits early.